File: 1260.cpp
#include <iostream>
#include <queue>
using namespace std;

int arr[1001][1001] = {0, };
bool visit[1001] = { false, };
int N ,M ,V;

void init(){
	for(int i = 0; i <= 1000; i++){
		visit[i] = false;
	}
}

void dfs(int s){

	cout << s << " ";
	visit[s] = true;

	for(int i = 1; i <= N; i++){
		if(arr[s][i] && !visit[i]){
			dfs(i);
		}
	}
}

void bfs(int s){
	queue<int> q;

	q.push(s);
	visit[s] = true;

	while(!q.empty()){
		int front = q.front();
		visit[front] = true;
		cout << front << " ";
		q.pop();

		for(int i = 1; i <= N; i++){
			if(arr[front][i] && !visit[i]){
				visit[i] = true;
				q.push(i);
			}
		}
	}

}


int main(){

	int src, dst;
	cin >> N >> M >> V;

	for(int i = 0; i < M; i++){
		cin >> src >> dst;
		arr[src][dst] = 1;
		arr[dst][src] = 1;
	}

	dfs(V);
	cout << "\n";

	init();
	bfs(V);

	return 0;
}

File: 1260.py
arr = [[0 for i in range(1001)] for i in range(1001)]
visit = [False for i in range(1001)]
n = 0
q = []

def init():
    global visit
    for i in range(1001):
        visit[i] = False

def dfs(num):
    global arr
    global visit
    global n
    print(num, end=' ')
    visit[num] = True
    for i in range(1, n+1):
        if arr[num][i] == 1 and visit[i] == False:
            dfs(i)

def bfs():
    global arr
    global visit
    global n
    global q

    while q:
        front = q[0]
        #visit[front] = True
        print(front, end=' ')
        del(q[0])

        for i in range(1, n+1):
            if arr[front][i] == 1 and visit[i] == False:
                visit[i] = True
                q.append(i)


def main():
    global arr
    global visit
    global n
    n, m, v = map(int, input().split())
    for i in range(m):
        src, dest = map(int, input().split())
        arr[src][dest] = 1
        arr[dest][src] = 1
    dfs(v)
    print()
    init()
    q.append(v)
    visit[v] = True
    bfs()
    print()

if __name__ == "__main__":
    main()